src/llama_stack_api/inference.py

```diff
@@ -556,6 +556,76 @@ class OpenAIResponseFormatJSONObject(BaseModel):
 register_schema(OpenAIResponseFormatParam, name="OpenAIResponseFormatParam")
 
 
+@json_schema_type
+class FunctionToolConfig(BaseModel):
+    name: str
+
+
+@json_schema_type
+class OpenAIChatCompletionToolChoiceFunctionTool(BaseModel):
+    """Function tool choice for OpenAI-compatible chat completion requests.
+
+    :param type: Must be "function" to indicate function tool choice
+    :param function: The function tool configuration
+    """
+
+    type: Literal["function"] = "function"
+    function: FunctionToolConfig
+
+    def __init__(self, name: str):
+        super().__init__(type="function", function=FunctionToolConfig(name=name))
+
+
+@json_schema_type
+class CustomToolConfig(BaseModel):
+    name: str
```

Comment:
should these be `Field`'s with defaults? or have a default in general?

Reply:
I don't know how we would set a default value here, since this represents the name of a function that needs to come from the user if they choose to force the model to use a specific custom tool in the first place